### PR TITLE
potentially invoke lfs for 'git update-index'

### PR DESCRIPTION
**Pull request by technoweenie**
Brings LFS back for `git update-index` calls. This fixes issues where `git status` is unclean after a `git lfs pull` or `git lfs clone`.

/cc https://github.com/git-lfs/git-lfs/issues/2646#issuecomment-334204675

### CODE REVIEW

File: test/test-clone.sh [hunk 1]
```diff
@@ -119,6 +123,7 @@ begin_test "cloneSSL"
 
   newclonedir="testcloneSSL1"
   git lfs clone "$SSLGITSERVER/$reponame" "$newclonedir" 2>&1 | tee lfsclone.log
+  #assert_clean_status
```
Comment (larsxschneider) [inline test/test-clone.sh]: Is this commented intentionally?

File: git/git.go [hunk 2]
```diff
@@ -459,7 +459,7 @@ func DefaultRemote() (string, error) {
 }
 
 func UpdateIndexFromStdin() *subprocess.Cmd {
-	return gitNoLFS("update-index", "-q", "--refresh", "--stdin")
+	return git("update-index", "-q", "--refresh", "--stdin")
```
Comment (larsxschneider) [inline git/git.go]: OK, here is my understanding:

We hit this code path *only* if we manually resolve the Git LFS pointers. Manually here means: we don't use the Git clean/smudge mechanism.  This happens in the `git lfs clone ...` and `git lfs pull` calls. In these calls the following happens:

1. Git checks out all files without running the Git LFS filters. We have the Git LFS pointer files checked out as a result.
2. Git LFS iterates over all pointer files and replaces them with the actual content
3. Git LFS asks Git to re-index the changed content (this happens with the `update-index` call above). During the re-index Git needs to `clean` the real content to calculate the pointer file and compare that against the version stored in the repository. This way `git status` can tell us if the file was changed or not.

--> I think it makes sense to run Git with the clean filter here.

**Comment by technoweenie** [comment]
Thanks for the review :metal:

**Comment by larsxschneider** [comment]
Your welcome! Do you know when 2.3.3 will be released with the fix?

**Comment by technoweenie** [comment]
I have some stuff this morning, but I could release it this afternoon. Though, I don't think a friday afternoon release is a good idea, so I'm leaning towards Monday. What do you think?

**Comment by larsxschneider** [comment]
I trust your judgement here. I agree with the Friday argument. However, this is a pretty nasty bug, too.

**Pull request merged**